peripherals/gpio/gpio_f4.yaml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 The gd32-rs authors.
+#
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"GPIO*":
+  CTL:
+    "CTL*":
+      Input: [0, "Input mode (reset state)"]
+      Output: [1, "General purpose output mode"]
+      Alternate: [2, "Alternate function mode"]
+      Analog: [3, "Analog mode"]
+  OMODE:
+    "OM*":
+      PushPull: [0, "Output push-pull (reset state)"]
+      OpenDrain: [1, "Output open-drain"]
```

Comment:
Again, it looks like a lot of this is just copied from `gpio.yaml`, please include it rather than copying it. Then this file can have only the parts that are specific to the F4 family.

Reply:
Same as for the ADC, will figure out the common parts in a future commit.